xarray/core/computation.py

```diff
@@ -1380,24 +1379,26 @@ def _calc_idxminmax(
     # This will run argmin or argmax.
     indx = func(array, dim=dim, axis=None, keep_attrs=keep_attrs, skipna=skipna)
 
-    # Get the coordinate we want.
-    coordarray = array[dim]
-
     # Handle dask arrays.
-    if isinstance(array, dask_array_type):
-        res = dask_array.map_blocks(coordarray, indx, dtype=indx.dtype)
+    if isinstance(array.data, dask_array_type):
+        res = indx.copy(
```

Comment:
Could you add a brief comment on why we need special logic for dask.array?

I guess it's because you can't index a NumPy array with a dask array?

Reply:
I'm a bit lost in reconstructing the error, but it seems to break when broadcasting indices here: https://github.com/pydata/xarray/blob/2c77eb531b6689f9f1d2adbde0d8bf852f1f7362/xarray/core/variable.py#L533-L536

I originally stumbled over this while testing the new idxmin/idxmax feature with dask backed arrays. I'll gladly add any comment per your recommendation.

Reply:
Strange, now it seems the 2D case works correct without the added dask-handling and only the 1D case is failing with the above error. :grimacing:

Reply:
@shoyer Good chance that I've been on the wrong track for this. I've removed the special dask treatment to see which tests are breaking and how to resolve.